ghcide/src/Development/IDE/Spans/Documentation.hs

```diff
@@ -63,7 +64,9 @@ lookupKind env mod =
     fmap (fromRight Nothing) . catchSrcErrors (hsc_dflags env) "span" . lookupName env mod
 
 getDocumentationTryGhc :: HscEnv -> Module -> Name -> IO SpanDoc
-getDocumentationTryGhc env mod n = head <$> getDocumentationsTryGhc env mod [n]
+getDocumentationTryGhc env mod n =
+    headNote "Development.IDE.Spans.Documentation.getDocumentationTryGhc"
```

Comment:
Remove

Reply:
This seems potentially kind of bad! I think we should do something here. Just return a `Maybe`?

Reply:
I think this is similar to `uses` in that we are local equational reasoning suffices to prove the absence of errors. You could generalise `getDocumentationsTryGhc` to take  `Traversable` and use a `NonEmpty` list in `getDocumentationTryGhc`